File: android/core/src/main/java/app/visly/shard/ShardContext.kt
package app.visly.shard

import android.content.Context
import android.content.ContextWrapper

internal interface ActionDelegate {
    fun on(action: String, value: JsonValue?)
}

class ShardContext(ctx: Context) : ContextWrapper(ctx) {
    internal var actionDelegate: ActionDelegate? = null

    fun dispatch(action: String, value: JsonValue?) {
        this.actionDelegate?.on(action, value)
    }
}

File: ios/ShardKit/Classes/ShardContext.swift
/**
 * Copyright (c) Visly Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

internal protocol ShardContextDelegate {
    func onActionDispatched(action: String, value: JsonValue?)
}

public class ShardContext {
    internal var delegate: ShardContextDelegate? = nil
    
    private(set) public var fontCollection: FontCollection
    
    public func dispatch(action: String, value: JsonValue?) {
        delegate?.onActionDispatched(action: action, value: value)
    }
    
    init(fontCollection: FontCollection = [:]) {
        self.fontCollection = fontCollection
    }
}
